app/assets/javascripts/widgets/notifications-badge.js

```diff
@@ -76,36 +83,34 @@
     };
 
     this.renderNotifications = function() {
-      self.dropdownNotifications.empty();
+      this.dropdownNotifications.find('.media.stream_element').remove();
       $.each(self.notifications, function(index, notifications) {
         $.each(notifications, function(index, notification) {
-          if($.inArray(notification, notifications) === -1)
+          if($.inArray(notification, notifications) === -1){
             self.dropdownNotifications.append(notification.note_html);
+          }
         });
       });
 
+      $(".unread-toggle .entypo").each(function(){
+        $(this).tooltip('destroy').tooltip();
+      });
```

Comment:
I'm not really sure about my code here, I suspect there's a more efficient way to achieve this in the above loop.